### PR TITLE
Edit profile separate

### PR DESCRIPTION
**Pull request by vinyll**

### CODE REVIEW

**Comment by vinyll** [comment]
Uploading a picture does not work yet.
Frontend should be setup again @fuentesloic ;)

**Comment by fuentesloic** [comment]
@vinyll Should we close this one ? according to #258

**Comment by vinyll** [comment]
We should keep this one. #258 if for articles. This one concerns profiles.
Things remaining:
- fix tests
- fix the frontend ;)
- get the image upload work again

**Comment by fuentesloic** [comment]
@vinyll I fixed python error (and so test) and conditional rendering in templates (user profile/edit).
We still have issues:

- I'm not able to `POST` any change on edit user profile
- As the html tags changed for input (not editable H1, p, ...) we have to add style

Quick questions:

- Did you remember anything about why we change tags and not keeping old one?
- Did you remember anything about the `POST` errors?

I didn't have enougth time to investigate today so next step tomorrow 😺

File: templates/user/profile.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {% extends 'layout.html' %}
 
 {% block title %}{{ user|capitalize }}'s profile{% endblock %}
-{% block description %}{{ user|capitalize }} is on Neomad with {{ articles.count() }} articles published and travels tracking {% if user.allow_community %}enabled{% else %}disabled{% endif %}.{% endblock %}
+{% block description %}{{ user|capitalize }} is on Neomad with {{ articles.count() }} articles published and {% if user.allow_community %}sharing with the community{% else %}in hight privacy{% endif %}.{% endblock %}
```
Comment (vinyll) [inline templates/user/profile.html:5]: you mean _with high privacy_, but we could actually say _a private profile_ instead

File: templates/user/profile.html [hunk 2]
```diff
@@ -27,205 +27,53 @@
         onerror="this.src='{{ url_for('static', filename='images/avatar.png') }}'">
     </figure>
     <div>
-      <h1 name=username>{{ user.username or 'Insert a username' }}</h1>
-      <p name=about>{{ user.about|htmlnewline or 'Introduce yourself here' }}</p>
+      <h1 name=username>{{ user.username }}</h1>
+      <p name=about>{{ user.about|htmlnewline }}</p>
```
Comment (vinyll) [inline templates/user/profile.html:31]: We need these defaults in order to be able to type in

File: templates/user/edit.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          <caption>
+            <input type=file name=avatar>
+            {% if user.avatar %}
+            <label><input type=checkbox name=delete value=1>Remove this picture</label>
+            {% endif %}
+          </caption>
+        </figure>
+
+        <section>
+          <input name=username value="{{ user.username or '' }}" placeholder="Insert a username">
```
Comment (vinyll) [inline templates/user/edit.html]: We need a default text in order to be able to type in: `{{ user.username or 'Insert your username' }}`

File: templates/user/edit.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          <textarea name=about placeholder="Tell a little about yourself">{{ user.about|htmlnewline }}</textarea>
```
Comment (vinyll) [inline templates/user/edit.html]: We need a defaut text in order to type in`{{ user.about|htmlnewline || "Tell a little about yourself" }}`
This is as a placeholder, indeed, but also to force a height.
Note that you can keep the placeholder anyways :)

File: user/views.py [hunk 5]
```diff
@@ -44,28 +45,47 @@ def privacy_delete_trip(date):
     return redirect('privacy'), 204
 
 
-@app.route('/profile/', methods=['patch'])
+@app.route('/profile/')
```
Comment (vinyll) [inline user/views.py:48]: Why do you remove the `methods=['patch']`? We can't post with `patch` anymore then.
How would post the changes?

**Review by vinyll — COMMENTED** [review]
Wouhou 👏  :))

File: templates/user/profile.html (hunk 1 above)
Comment (fuentesloic) [reply templates/user/profile.html:5]: as you made this comment on your own code I'm not sure about what I have to do with this 😛

File: templates/user/profile.html (hunk 2 above)
Comment (fuentesloic) [reply templates/user/profile.html:31]: as you made this comment on your own code I'm not sure about what I have to do with this 😛

File: user/views.py (hunk 5 above)
Comment (fuentesloic) [reply user/views.py:48]: as you made this comment on your own code I'm not sure about what I have to do with this 😛

File: templates/user/profile.html (hunk 2 above)
Comment (fuentesloic) [reply templates/user/profile.html:31]: as you made this comment on your own code I'm not sure about what I have to do with this 😛

**Comment by fuentesloic** [comment]
Here a quick style update, don't be too picky as I'm going to get done the styleguide
Screenshot_2019-12-07 My profile
Screenshot_2019-12-07 My profile(1)

**Comment by vinyll** [comment]
That's all very cool! LGTM 🚀 
PS: you are the one who need to validate this PR ;)

**Comment by fuentesloic** [comment]
@vinyll I didn't validate it as some comments are still on-going, are they irrelevant?

**Comment by vinyll** [comment]
I don't see any comment left.
Mergeable to me

**Comment by vinyll** [comment]
uploading pics work. Frontend fixed. tests pass.

**Review by fuentesloic — APPROVED** [review]
🚀

**Pull request merged**